python/mlx/nn/losses.py

```diff
@@ -195,6 +220,32 @@ def kl_div_loss(
     return _reduce(loss, reduction)
 
 
+def smooth_l1_loss(
+    predictions: mx.array, targets: mx.array, beta: float = 1.0, reduction: str = "mean"
+) -> mx.array:
+    """
+    Calculate the Smooth L1 Loss between predictions and true values.
+
+    Args:
+        predictions (mx.array): Predicted values.
+        targets (mx.array): Ground truth values.
+        beta (float, optional): The threshold at which to change from L2 to L1 loss. Defaults to 1.0.
+        reduction (str, optional):  reduction (str, optional): Specifies the reduction to apply to the output:
+          ``'none'`` | ``'mean'`` | ``'sum'``. Default: ``'mean'``.
+
+    Returns:
+        mx.array: The calculated loss. It will be a single element array if reduction is 'mean' or 'sum',
+                     or the same shape as predictions and targets if reduction is 'none'.
+    """
+
+    diff = mx.abs(predictions - targets)
+    loss = mx.where(
+        diff < beta, 0.5 * mx.square(predictions - targets) / beta, diff - 0.5 * beta
```

Comment:
do you mean using 
```mx.maximum(predictions - targets, targets - predictions) ```
instead of
```mx.abs(predictions - targets)``` ? 

Could you explain why mx.maximum would be faster?

Reply:
No that's not what I mean sorry for not being clear:
```
 diff = predictions - targets
 loss = mx.maximum(0.5 * mx.square(diff) / beta, mx.abs(diff) - 0.5 * beta)
```

Reply:
The smooth L1 loss is defined according to the piecewise function below, I don't think taking the maximum would be correct here. What do you think ? 

$$
\text{loss} = 
\begin{cases} 
\frac{0.5 \times (predictions - targets)^2}{beta}, & \text{if } diff < beta \\
diff - 0.5 \times beta, & \text{otherwise}
\end{cases}
$$

Reply:
Sorry that's my bad.. disregard the maximum suggestion. Storing the difference is still a good idea.

```
    diff = predictions - targets
   abs_diff = mx.abs(diff)
    loss = mx.where(
        abs_diff < beta, 0.5 * mx.square(diff) / beta, abs_diff - 0.5 * beta
```